adam-core/src/main/scala/org/bdgenomics/adam/rdd/contig/NucleotideContigFragmentRDD.scala

```diff
@@ -95,14 +95,18 @@ private[rdd] object NucleotideContigFragmentRDD extends Serializable {
  * A wrapper class for RDD[NucleotideContigFragment].
  * NucleotideContigFragmentRDD extends ReferenceFile. To specifically access a ReferenceFile within an RDD,
  * refer to:
+ *
  * @see ReferenceContigMap
  *
  * @param rdd Underlying RDD
  * @param sequences Sequence dictionary computed from rdd
  */
 case class NucleotideContigFragmentRDD(
     rdd: RDD[NucleotideContigFragment],
-    sequences: SequenceDictionary) extends AvroGenomicRDD[NucleotideContigFragment, NucleotideContigFragmentRDD] with ReferenceFile {
+    sequences: SequenceDictionary,
+    optPartitionMap: Option[Array[Option[(ReferenceRegion, ReferenceRegion)]]] = None) extends AvroGenomicRDD[NucleotideContigFragment, NucleotideContigFragmentRDD] with ReferenceFile {
```

Comment:
I like default method parameters less as time goes by.  Could this be redone as multiple apply methods in an object and full parameters in the case class constructor?
```scala
private[rdd] object FooRDD extends Serializable {                                                                    
  def apply(rdd: RDD[Foo]): FooRDD): FooRDD = {
  }
  def apply(rdd: RDD[Foo], sequences: SequenceDictionary): FooRDD = {
  }
  /* ... other useful combinations of parameters ... */
}

case class FooRDD(
  rdd: RDD[Foo],
  sequences: SequenceDictionary,
  optPartitionMap: Option[...],
  /* ...all of the parameters... */) extends AvroGenomicRDD[Foo} ... {

}
```

See
https://github.com/databricks/scala-style-guide#java-default-param-values
https://github.com/databricks/scala-style-guide#apply-method